packages/netlify-cms-core/src/backend.ts

```diff
@@ -411,24 +337,42 @@ export class Backend {
   }
 
   listEntries(collection: Collection) {
-    const listMethod = this.implementation[selectListMethod(collection)];
     const extension = selectFolderEntryExtension(collection);
-    return listMethod
-      .call(this.implementation, collection, extension)
-      .then((loadedEntries: ImplementationEntry[]) => ({
-        entries: this.processEntries(loadedEntries, collection),
-        /*
+    let listMethod: () => Promise<ImplementationEntry[]>;
+    const collectionType = collection.get('type');
+    if (collectionType === FOLDER) {
+      listMethod = () =>
+        this.implementation.entriesByFolder(
+          collection.get('folder') as string,
```

Comment:
Getting the folder and the depth was duplicated in each backend so I moved it here